specification/servicefabric/resource-manager/Microsoft.ServiceFabric/2017-07-01-preview/servicefabric.json

```diff
@@ -0,0 +1,3268 @@
+        ],
+        "x-ms-examples": {
+          "List cluster versions": {
+            "$ref": "examples/ClusterVersionsList_example.json"
+          }
+        },
+        "tags": [
+          "ClusterVersion"
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - List cluster code versions successfully",
+            "schema": {
+              "$ref": "#/definitions/ClusterCodeVersionsListResult"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ServiceFabric/locations/{location}/environments/{environment}/clusterVersions": {
+      "get": {
+        "operationId": "ClusterVersions_ListByEnvironment",
+        "summary": "List cluster code versions by environment",
+        "description": "List cluster code versions by environment\n",
+        "parameters": [
+          {
+            "$ref": "#/parameters/locationForClusterCodeVersions"
+          },
+          {
+            "$ref": "#/parameters/environment"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          }
+        ],
+        "x-ms-examples": {
+          "List cluster versions by environment": {
+            "$ref": "examples/ClusterVersionsListByEnvironment.json"
+          }
+        },
+        "tags": [
+          "ClusterVersion"
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - List cluster code versions successfully",
+            "schema": {
+              "$ref": "#/definitions/ClusterCodeVersionsListResult"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.ServiceFabric/operations": {
+      "get": {
+        "tags": [
+          "Operations"
+        ],
+        "summary": "Lists all of the available ServiceFabric Resource Manager REST API operations.",
+        "description": "",
+        "operationId": "Operations_List",
+        "parameters": [
+          {
+            "name": "api-version",
+            "in": "query",
+            "description": "The version of the ServiceFabric resouce provider api",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/OperationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorModel"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceFabric/clusters/{clusterName}/applicationTypes/{applicationTypeName}": {

```

Comment:
Isnt applicationTypes a proxy resource? Proxy resources dont have a location, but in the response of this API, i see you have a location. Is it a tracked resource?

Reply:
We were planning to make it a tracked resource, but decided against it. We left this as is in case we changed our mind.